### PR TITLE
Simplify addmetadata.

### LINKED ISSUE #273: Revisit validation package pkg/validate
As per [discussion while adding edit add label/annotation](https://github.com/kubernetes-sigs/kustomize/pull/147#discussion_r211325317), a validation package for input to match syntax was added.

This package currently has rudimentary placeholder validation. Ultimately, it should reflect api-machinery standards (pending agreement/discussion).

@monopole any additional thoughts/input?

### PR DESCRIPTION
**Pull request by monopole**
Factored code into shorter methods.
Dropped unnecessary branching and variable passing.
Removed dependence on apimachinery in test - used fake injection.
Dropped unnecessary printing of nil errors.
Added more tests.

Fixes #273

### CODE REVIEW

File: pkg/commands/addmetadata_test.go [hunk 1]
```diff
@@ -29,82 +30,120 @@ func TestRunAddAnnotation(t *testing.T) {
 	var o addMetadataOptions
 	o.metadata = map[string]string{"owls": "cute", "otters": "adorable"}
 
-	err := o.RunAddAnnotation(fakeFS, annotation)
+	err := o.RunAddAnnotation(fakeFS)
 	if err != nil {
-		t.Errorf("unexpected error: could not write to kustomization file")
+		t.Errorf("incorrect error: could not write to kustomization file")
```
Comment (Liujingfang1) [inline pkg/commands/addmetadata_test.go]: I think `unexpected error` is better than `incorrect error` here.

File: pkg/commands/addmetadata_test.go [hunk 2]
```diff
@@ -115,82 +154,107 @@ func TestRunAddLabel(t *testing.T) {
...
 	fakeFS.WriteFile(constants.KustomizationFileName, []byte(kustomizationContent))
-	cmd := newCmdAddLabel(fakeFS)
+	v := validators.MakeHappyMapValidator(t)
+	cmd := newCmdAddLabel(fakeFS, v.Validator)
 	args := []string{"no,value:"}
 	err := cmd.RunE(cmd, args)
+	v.VerifyCall()
 	if err != nil {
-		t.Errorf("unexpected error: %v", err.Error())
+		t.Errorf("incorrect error: %v", err.Error())
```
Comment (Liujingfang1) [inline pkg/commands/addmetadata_test.go]: And this line, too.

Comment (monopole) [reply pkg/commands/addmetadata_test.go]: got it

**Pull request merged**